gateway/metrics/metrics.go

```diff
@@ -81,6 +82,15 @@ func BuildMetricsOptions() MetricOptions {
 		[]string{"method", "path", "status"},
 	)
 
+	startedCounter := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: "function",
```

Comment:
I think the opts are more correct if we use 

```go
prometheus.CounterOpts{
    Namespace: "function",
    Subsystem: "requests",
    Name: "started_total",
    Help: "The total number of function HTTP requests started.",
}
```

It should result in the same metric: `function_requests_started_total` per https://godoc.org/github.com/prometheus/client_golang/prometheus#Opts

Reply:
@stefanprodan what would you say? I'm happy to take a recommendation as long as its consistent with what we have. Does namespace/subsystem/name have an effect on how metrics are stored? What did we do with the previous metrics?

Reply:
it only impacts how the function is named, it basically treats namespace, subsystem, and name as parts that are joined together.  This just uses the options more semantically and improves the long term readability, imo.  For example, if we wanted to add a counter for sync requests it would then look like 

```go
prometheus.CounterOpts{
    Namespace: "function",
    Subsystem: "async_requests",
    Name: "started_total",
    Help: "The total number of async function HTTP requests started.",
}
```
it should be easier to verify things like spelling mistakes etc because each individual string is shorter

Reply:
What if we had one counter and then added a label like async: true/false or 1/0?

Reply:
Does the suggestion give a consistent name? Wouldn't we want to match with this?

```
	gatewayFunctionInvocation := prometheus.NewCounterVec(
		prometheus.CounterOpts{
			Name: "gateway_function_invocation_total",
			Help: "Individual function metrics",
		},
		[]string{"function_name", "code"},
	)
```

i.e. 

```
prometheus.CounterOpts{
    Namespace: "gateway",
    Subsystem: "function",
    Name: "started_total",
...
```

?

I might be missing something.

Reply:
adding another label makes sense to me, especially since it only has two values, it is equivalent to creating two metrics anyway.

Reply:
To your point about 

```go
	gatewayFunctionInvocation := prometheus.NewCounterVec(
		prometheus.CounterOpts{
			Name: "gateway_function_invocation_total",
			Help: "Individual function metrics",
		},
		[]string{"function_name", "code"},
	)
```

I don't think the current metric in this diff has a gateway prefix does it?  I expect code in the current diff to produce `function_requests_started_total`

Reply:
It does have the namespace yes, maybe it changed since you looked at the diff last?

<img width="1039" alt="Screenshot 2020-03-12 at 10 45 38" src="https://user-images.githubusercontent.com/6358735/76513734-a2379600-644e-11ea-96e1-e6d6c0d75a3f.png">

Reply:
yeah, looks good. The github UI seems to be wonky for me today.  It seems to be stateful with commit selections sometimes